packages/esm-patient-registration-app/src/patient-registration/patient-registration.component.tsx

```diff
@@ -126,123 +131,62 @@ export const PatientRegistration: React.FC<PatientRegistrationProps> = ({ savePa
-                });
-              });
-              break;
-            default:
-              if (prop === 'country' || prop === 'postalCode') {
-                initialAddressFieldValues[prop] = address[prop];
-              }
-          }
-        });
-      }
-
-      if (patient.deceasedBoolean || patient.deceasedDateTime) {
-        initialFormValues.isDead = true;
-        initialFormValues.deathDate = patient.deceasedDateTime ? patient.deceasedDateTime.split('T')[0] : '';
-      }
+    if (patient) {
+      setPatientUuidMap({ ...patientUuidMap, ...getPatientUuidMapFromFhirPatient(patient) });
+      setInitialAddressFieldValues({ ...initialAddressFieldValues, ...getAddressFieldValuesFromFhirPatient(patient) });
+      setInitialFormValues({ ...initialFormValues, ...getFormValuesFromFhirPatient(patient) });
 
       const abortController = new AbortController();
       fetchPatientPhotoUrl(patient.id, config.concepts.patientPhotoUuid, abortController).then((value) =>
         setCurrentPhoto(value),
-      ); // TODO: Edit mode
+      );
 
       return () => abortController.abort();
     }
-  }, [inEditMode]);
+  }, [patient]);
 
   useEffect(() => {
     for (const patientIdentifier of patientIdentifiers) {
       if (!initialFormValues[patientIdentifier.fieldName]) {
-        initialFormValues[patientIdentifier.fieldName] = '';
+        setInitialFormValues({ ...initialFormValues, [patientIdentifier.fieldName]: '' });
       }
 
-      initialFormValues['source-for-' + patientIdentifier.fieldName] =
-        patientIdentifier.identifierSources.length > 0 ? patientIdentifier.identifierSources[0].name : '';
+      setInitialFormValues({
+        ...initialFormValues,
+        ['source-for-' + patientIdentifier.fieldName]:
+          patientIdentifier.identifierSources.length > 0 ? patientIdentifier.identifierSources[0].name : '',
+      });
     }
   }, [patientIdentifiers]);
 
-  useEffect(() => {
-    if (capturePhotoProps?.base64EncodedImage || capturePhotoProps?.imageFile) {
-      setCurrentPhoto(capturePhotoProps.base64EncodedImage || URL.createObjectURL(capturePhotoProps.imageFile));
-    }
-  }, [capturePhotoProps]);
-
   useEffect(() => {
     const addressTemplateXml = addressTemplate.results[0].value;
     if (!addressTemplateXml) {
       return;
     }
 
     const { addressFieldValues, addressValidationSchema } = parseAddressTemplateXml(addressTemplateXml);
+    setValidationSchema((validationSchema) => validationSchema.concat(addressValidationSchema));
 
-    for (const { name, defaultValue } of addressFieldValues) {
-      if (!initialAddressFieldValues[name]) {
-        initialAddressFieldValues[name] = defaultValue;
+    // `=== false` is here on purpose (`inEditMode` is a triple state value).
+    // We *only* want to set initial address field values when *creating* a patient.
+    // We must wait until after loading for this info.
+    if (inEditMode === false) {
```

Comment:
This is the only exception to my previous comment.
Overwriting patient address data from the backend with _default_ values seemed wrong to me. I can remove the `if` here though if that's a wrong assumption on my side.